### PR TITLE
Adding ssl support back into 0.4.0

### PR DESCRIPTION
**Pull request by jloveland**
adding ssl back in, moving scripts to script folder so we can ./script/gereratate-ssl-certs.sh

### CODE REVIEW

File: config/lib/socket.io.js [hunk 1]
```diff
@@ -2,19 +2,36 @@
 
 // Load the module dependencies
 var config = require('../config'),
-	path = require('path'),
+		path = require('path'),
```
Comment (ilanbiala) [inline config/lib/socket.io.js]: indentation is off.

File: config/lib/socket.io.js [hunk 2]
```diff
@@ -2,19 +2,36 @@
...
-    var server = http.createServer(app);
+	var server;
+	if (config.secure === true) {
+		// Load SSL key and certificate
+		var privateKey = fs.readFileSync('./config/sslcerts/key.pem', 'utf8');
+		var certificate = fs.readFileSync('./config/sslcerts/cert.pem', 'utf8');
 
+		// Create new HTTPS Server
+		server = https.createServer({
+				key: privateKey,
```
Comment (ilanbiala) [inline config/lib/socket.io.js]: indentation and formatting

**Comment by ilanbiala** [comment]
@jloveland why `script` and not `scripts`? @lirantal any issue with merging this in?

**Comment by jloveland** [comment]
@ilanbiala, this is just a preference..most projects I work on put all the scripts into a `script` folder, mostly inspired by [holman's dotfiles](https://github.com/holman/dotfiles/tree/master/script). However, I can change `script` back to `scripts` if you want. let me know.

**Comment by ilanbiala** [comment]
@jloveland honestly I think it just comes down to it rolling off the tongue easier when telling other people and it seeming more natural because it's a folder of scripts.

**Comment by jloveland** [comment]
@ilanbiala I changed `script` back to `scripts`

**Comment by ilanbiala** [comment]
@lirantal LGTY?

**Comment by lirantal** [comment]
looks ok

**Pull request merged**